cli/compose/loader/loader.go

```diff
@@ -349,24 +349,20 @@ func createTransformHook(additionalTransformers ...Transformer) mapstructure.Dec
 
 // keys needs to be converted to strings for jsonschema
 func convertToStringKeysRecursive(value any, keyPrefix string) (any, error) {
-	if mapping, ok := value.(map[any]any); ok {
+	if mapping, ok := value.(map[string]any); ok {
 		dict := make(map[string]any)
 		for key, entry := range mapping {
-			str, ok := key.(string)
-			if !ok {
-				return nil, formatInvalidKeyError(keyPrefix, key)
-			}
```

Comment:
But while it makes sense too use `map[string]any` (i.e., YAML keys would be a string, always, even it it looks like an `int`  or a weird string like `{object}:`, that also means that we can't produce custom errors for those ("you used  an integer as key, did you mean something else?").